R/error-checks.R

```diff
@@ -1,33 +1,51 @@
+isSameLength <- function(...) {
+  args <- list(...)
+  nrOfLengths <- length(unique(lengths(args)))
+
+  return(nrOfLengths == 1)
+}
+
 #' Check if the provided object is of certain type
 #'
 #' @param object An object or a list of objects
-#' @param type String representation of the type that should be checked for
+#' @param type String  representation or Class of the type that should be checked for
 #'
 #' @return TRUE if the object or all objects inside the list are of the given type.
 #' Only the first level of the given list is considered.
-#'
-#'
 isOfType <- function(object, type) {
   if (is.null(object)) {
     return(FALSE)
   }
 
   object <- c(object)
-  isSameType <- all(sapply(
-    object,
-    function(x) inherits(x, type)
-  ))
+  type <- typeNamesFrom(type)
 
-  return(isSameType)
+  paste(type)
+  all(sapply(object, function(x) inherits(x, type)))
 }
 
-isSameLength <- function(...) {
-  args <- list(...)
-  nrOfLengths <- length(unique(lengths(args)))
+validateIsOfType <- function(object, type, nullAllowed = FALSE) {
+  if (nullAllowed && is.null(object)) {
+    return()
+  }
 
-  return(nrOfLengths == 1)
+  if (isOfType(object, type)) {
+    return()
+  }
+  # Name of the variable in the calling function
+  objectName <- deparse(substitute(object))
+  objectTypes <- typeNamesFrom(type);
+
+  stop(messages$errorWrongType(objectName, class(object)[1], objectTypes))
 }
 
+typeNamesFrom <- function(type){
+  if (is.character(type)) {
+    return(type)
+  }
+  type <- c(type)
+  sapply(type, function(t) t$classname)

```

Comment:
This is just an extension that allows to either pass the class name or the class itself
e,g, validateIfOfType(obj, "Simulation") or validateIsOfType(obj, Simulation)